File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] arr) {
        
        HashMap<Integer, Integer> hm=new HashMap<>();
        

        for(int i=0;i<arr.length;i++)
        {
            if(hm.containsKey(arr[i]))
                hm.put(arr[i], hm.get(arr[i])+1);
            else
                hm.put(arr[i],1);
        }
        int l=arr.length/2;
        for(int i=0;i<arr.length;i++)
        {
            if(hm.get(arr[i])>l)
                return arr[i];
        }
        return -1;
    }
}

/*

int c=0,ele=0;

		for(int i=0;i<arr.length;i++)
		{
			if(c==0)
			ele=arr[i];
			if(ele==arr[i])
			c++;
			else
			c--;
		}
		
		return ele;
        
        */

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& arr) {
        
        unordered_map<int,int> hm;
        
        int n=arr.size();
        
        for(int i=0;i<n;i++)
        {
            if(hm.find(arr[i])==hm.end())
                hm[arr[i]]=1;
            else
                hm[arr[i]]=hm[arr[i]]+1;
        }
        
        int no_of_occ=n/2;
        
        for(int i=0;i<hm.size();i++)
        {
            if(hm[i]>no_of_occ)
                return i;
        }
        
        return -1;
        
    }
};